### PR TITLE
[MuJoCo] move `camera_id` state to inside the rendering class

### PR DESCRIPTION
**Pull request by Kallinteris-Andreas**
# Description
`MujocoEnv` has the member variable `camera_id` and `mujoco_renderer`.
This PR moves the `MujocoEnv.camera_id` member variable inside `MujocoEnv.mujoco_renderer`

## motivation
all render related state should be inside the `mujoco_renderer` not the base class

## Type of change
- [ ] Documentation only change (no code changed)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:
- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: gymnasium/envs/mujoco/mujoco_env.py [hunk 1]
```diff
@@ -360,7 +360,7 @@ def __init__(
         height: int = DEFAULT_SIZE,
         camera_id: Optional[int] = None,
         camera_name: Optional[str] = None,
-        default_camera_config: Optional[Dict[str, Union[float, int]]] = None,
+        default_camera_config: Optional[Dict[str, Union[float, int]]] = {},
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/envs/mujoco/mujoco_env.py:363]: Is there a reason why this was changed from `None`. 
It is bad practice to have mutable types as arguments

Comment (Kallinteris-Andreas) [reply gymnasium/envs/mujoco/mujoco_env.py]: @pseudo-rnd-thoughts  reverted

**Review by pseudo-rnd-thoughts — APPROVED** [review]

File: gymnasium/envs/mujoco/mujoco_env.py (hunk 1 above)
Comment (pseudo-rnd-thoughts) [reply gymnasium/envs/mujoco/mujoco_env.py]: Thanks for changing

**Comment by Kallinteris-Andreas** [comment]
Forgot to mention validation can be found here: https://github.com/Kallinteris-Andreas/gymnasium-changes-validation/tree/main/camera_id_refactor
the `"depth_array"` and `"rgb_array"` were validated to be the same for 1 mil steps, and no measurable slowdown,
`"human"` was not changed for this code, so a single check was done manually.

This should not affect any of the libraries that use `MujocoEnv` since the class abstracts it anyway

**Pull request merged**